apigee_sync.go

```diff
@@ -52,41 +52,33 @@ func postPluginDataDelivery(e apid.Event) {
 }
 
 /*
- * Helper function that sleeps for N seconds if comm with change agent
- * fails. The retry interval gradually is incremented each time it fails
- * till it reaches the Polling Int time, and after which it constantly
- * retries at the polling time interval
+ * Polls change agent for changes. In event of errors, uses a doubling
+ * backoff from 200ms up to a max delay of the configPollInterval value.
  */
 func updatePeriodicChanges() {
 
-	times := 1
-	pollInterval := config.GetInt(configPollInterval)
+	var backOffFunc func()
+	pollInterval := config.GetDuration(configPollInterval)
 	for {
-		startTime := time.Second
+		start := time.Now().Second()
```

Comment:
In time.go:
"Second returns the second offset within the minute specified by t, in the range [0, 59]."
Maybe we should use time.now().Unix(), or just use time.Now() with duration comparison.